libnavigation-ui/build.gradle

```diff
@@ -52,9 +53,9 @@ dependencies {
   ktlint dependenciesList.ktlint
 
   // Navigation SDK
-  api project(':libnavigation-core')
-  implementation project(':libnavigation-util')
-  implementation project(':libtrip-notification')
+  api "com.mapbox.navigation:core:1.0.0-qa.4"
```

Comment:
`examples` app build is ❌ because of this

```
> Task :examples:multiDexListDebug FAILED
D8: Program type already present: com.******.navigation.metrics.BuildConfig

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':examples:multiDexListDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Error while merging dex archives: 
     Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
     Program type already present: com.******.navigation.metrics.BuildConfig
```

We're including `libnavigation-ui` that includes this once and also https://github.com/mapbox/mapbox-navigation-android/blob/39ef2ed6fb56d14105b978672d095d26c6a2f840/examples/build.gradle#L87 and hence the clash.

Can't we do something similar to what we do in https://github.com/mapbox/mapbox-events-android? 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2568#discussion_r391875671 It's depending on `project()` instead of a concrete artifact. What are the blockers? cc @harvsu

Reply:
In any case, if we want to move forward with this approach, could we extract these dependencies into `dependencies.gradle`?